### PR TITLE
Fixing empty PipeCwd issue.

### PR DESCRIPTION
**Pull request by rajkumar42**
PipeCwd will be null if the provided PipeCwd is not absolute.

### CODE REVIEW

**Comment by rajkumar42** [comment]
@gregg-miskelly

File: src/MICore/LaunchOptions.cs [hunk 1]
```diff
@@ -64,6 +64,14 @@ public enum LaunchCompleteCommand
     /// </summary>
     public sealed class PipeLaunchOptions : LaunchOptions
     {
+        /// <summary>
+        /// Creates an instance of PipeLaunchOptions
+        /// </summary>
+        /// <param name="pipePath">Path of the pipe program</param>
```
Comment (gregg-miskelly) [inline src/MICore/LaunchOptions.cs]: Might want to add something here, and to the xsd file (line 292) something like --

This should either be a full path to an executable or the name of a program on the PATH.

File: src/MICore/Transports/PipeTransport.cs [hunk 2]
```diff
@@ -93,9 +93,11 @@ public override void InitStreams(LaunchOptions options, out StreamReader reader,
         {
             PipeLaunchOptions pipeOptions = (PipeLaunchOptions)options;
 
+            string workingDirectory = pipeOptions.PipeCwd;
             if (!LocalLaunchOptions.CheckDirectoryPath(pipeOptions.PipeCwd))
```
Comment (gregg-miskelly) [inline src/MICore/Transports/PipeTransport.cs]: The one problem with this is that if the user provides a 'PipeCmd' attribute which is bogus, we will no longer throw. We might want to instead remove LaunchOptions.cs(82) and change this line to be:

```csharp
    if (!string.IsNullOrEmpty(workingDirectory))
    {
       if (!CheckDirectoryPath()..) throw;
    }
   else
   {
      workingDirectory = Path.GetDirectoryName(pipePath);
      if (!CheckDirectoryPath()..)
      {
        your new code
      }
```

**Comment by gregg-miskelly** [comment]
>             }

I think you want to remove this too

---
Refers to: src/MICore/LaunchOptions.cs:91 in 40c92ab. [](commit_id = 40c92ab6a3a0c6b652d94b90cc3b9d47086c5be4, deletion_comment = False)

**Comment by gregg-miskelly** [comment]
:shipit:

**Comment by rajkumar42** [comment]
>             }

yep! thanks

---
In reply to: [249264922](https://github.com/Microsoft/MIEngine/pull/453#issuecomment-249264922) [](ancestors = 249264922)

---
Refers to: src/MICore/LaunchOptions.cs:91 in 40c92ab. [](commit_id = 40c92ab6a3a0c6b652d94b90cc3b9d47086c5be4, deletion_comment = False)

**Pull request merged**